tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerGraph.java

```diff
@@ -340,6 +342,14 @@ private void validateHomogenousIds(final List<Object> ids) {
         }
     }
 
+    @Override
```

Comment:
A touch of javadoc here to explain that this provide a deep clone of the graph/vertices/edges and preserves ids.